### PR TITLE
Changes for migrating codecov-actions to v4

### LINKED ISSUE #1053: Code Coverage report upload issues with codecov-actionsv3
## Description
In recent commits we can see there is a error in code coverage reports upload. The issue tells about the missing token value.

## Expected Behavior
Update the codecov-actions from v3 to v4, Add codecov.io token in the ENV variables.

### PR DESCRIPTION
**Pull request by kedarkhaire**
Fixes #1053

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -70,14 +70,14 @@ jobs:
...
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       with:
         path: ${{ steps.composercache.outputs.dir }}
         key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
         restore-keys: ${{ runner.os }}-composer-
 
     - name: Checkout apigee_edge module
-      uses: actions/checkout@v3
+      uses: actions/checkout@main
```
Comment (shishir-intelli) [inline .github/workflows/php.yml]: why not v4?

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -150,16 +150,19 @@ jobs:
...
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
         name: browser-output-${{ matrix.php-version }}-${{ matrix.drupal-core }}-${{ matrix.instance-type }}-artifact
         path: drupal/sites/simpletest/browser_output/*
 
     - name: Upload coverage to Codecov
       if: ${{ matrix.drupal-core == '10.1.x' && matrix.php-version == '8.1' }}
-      uses: codecov/codecov-action@v3
+      uses: codecov/codecov-action@v4.3.1
```
Comment (shishir-intelli) [inline .github/workflows/php.yml]: why are we restricting to 4.3.1?

Comment (kedarkhaire) [reply .github/workflows/php.yml]: That is the latest release they have

File: .github/workflows/php.yml (hunk 1 above)
Comment (kedarkhaire) [reply .github/workflows/php.yml]: As per the codecov-actions examples, they used main branch for checkout, so added the same

Comment (shishir-intelli) [reply .github/workflows/php.yml]: can we please change it to v4

File: .github/workflows/php.yml [hunk 3]
```diff
@@ -150,16 +150,19 @@ jobs:
...
       with:
+        working-directory: drupal/modules/contrib/apigee_edge
```
Comment (shishir-intelli) [inline .github/workflows/php.yml]: no need of this line.

File: .github/workflows/php.yml [hunk 4]
```diff
@@ -150,16 +150,19 @@ jobs:
...
         files: /tmp/coverage_${{ matrix.instance-type }}.xml
         name: codecov-umbrella
+        codecov_yml_path: codecov.yml
```
Comment (shishir-intelli) [inline .github/workflows/php.yml]: Not required

Comment (kedarkhaire) [reply .github/workflows/php.yml]: Removed the line.

File: .github/workflows/php.yml (hunk 3 above)
Comment (kedarkhaire) [reply .github/workflows/php.yml]: Removed the line.

**Review by shishir-intelli — APPROVED** [review]
LGTM

**Pull request merged**